acs-category-theory-notes.tex

```diff
@@ -1644,4 +1652,379 @@ \section{Exercises}
   at $\catmon$, $\catpreo$, $\catpred$, and $\catrel$.
 \end{exercise}
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\chapter{Morphisms}
+\lecturedetails{19 October 2017}{M Fiore, S Lau, D Szamozvancev}
+
+\section{Isomorphisms}
+\begin{definition}[Isomorphism]
+An arrow $f : A \morpharrow B$ is an \emph{isomorphism} if there exists an
+inverse arrow $g : B \morpharrow A$ such that
+\[
+    g \comp f = \id{A} \quad \text{ and } \quad f \comp g = \id{B}
+\]
+We say $A$ and $B$ are \emph{isomorphic} if there exists an isomorphism between
+them.
+\end{definition}
+
+\begin{remark}
+If $f$ has an inverse $g$, then the inverse is unique and we write $g = f^{-1}$.
+\end{remark}
+
+\begin{proof}
+Suppose $h$ is also an inverse, so we have
+$h \comp f = \id{A}$ and $f \comp h = \id{B}$.
+Then
+\begin{alignat*}{4}
+    h \comp (f \comp g) &= h \comp \id{B} &&= h \quad \text{and} \\
+    (h \comp f) \comp g &= \id{A} \comp g &&= g.
+\end{alignat*}
+By associativity, $g = h$.
+\end{proof}
+
+\begin{example}
+In $\catset$, $f$ is an isomorphism iff it is a bijection, where
+\[
+    f : A \rightarrow B \text{ is a bijection } \iff
+    \forall b \in B.\ \exists ! a \in A.\ f(a) = b.
+\]
+
+A bijection is an \emph{injection} and a \emph{surjection}.
+
+\begin{minipage}[t]{0.50\textwidth}
+    \begin{center}
+    Injection\\
+    $\forall a, a' \in A.\ f(a) = f(a') \Rightarrow a = a'$
+
+    \begin{tikzpicture}
+        [
+            group/.style={ellipse, draw, minimum height=50pt,
+                          minimum width=30pt, label=above:#1},
+            dot/.style={circle, fill, minimum width=2.5pt, inner sep=0pt}
+        ]
+        \node (a) [dot] {};
+        \node (b) [dot, below=of a] {};
+        \node (p) [dot, right=50pt of a] {};
+        \node (q) [dot, below=of p] {};
+        \node (r) [dot, right=10pt of p] {};
+        \node (s) [dot, below=of r] {};
+        \node (t) [dot] at ($(r)!1/2!(s)$) {};
+        \foreach \i/\j in {a/p,b/q}
+            \draw [->, >=stealth,scale=3] (\i) -- (\j);
+        \node [fit=(a) (b), group=A] {};
+        \node [fit=(p) (q) (r) (s) (t), group=B] {};
+    \end{tikzpicture}
+    \end{center}
+\end{minipage}%
+\begin{minipage}[t]{0.50\textwidth}
+    \begin{center}
+    Surjection\\
+    $\forall b \in B.\ \exists a \in A.\ f(a) = b$
+
+    \vspace{15pt}
+    \begin{tikzpicture}
+        [
+            group/.style={ellipse, draw, minimum height=20pt,
+                          minimum width=50pt, label=left:#1},
+            dot/.style={circle, fill, minimum width=2.5pt, inner sep=0pt}
+        ]
+        \node (a) [dot] {};
+        \node (b) [dot, right=10pt of a] {};
+        \node (c) [dot, right=10pt of b] {};
+        \node (d) [dot, right=10pt of c] {};
+        \node (e) [dot, right=10pt of d] {};
+        \node (p) [dot, below=of b] {};
+        \node (q) [dot, below=of d] {};
```

Comment:
perhaps it is slightly confusing to have one of the diagrams horizontal and the other vertical?